File: flutter_ui/lib/core/services/userOperationClaimService.dart
import 'dart:convert';

import 'package:flutter_ui/core/interceptors/authInterceptor.dart';
import 'package:flutter_ui/core/models/deleteModel.dart';
import 'package:flutter_ui/core/models/response/listResponseModel.dart';
import 'package:flutter_ui/core/models/response/responseModel.dart';
import 'package:flutter_ui/core/models/response/singleResponseModel.dart';
import 'package:flutter_ui/core/models/userOperationClaim/userOperationClaimAddModel.dart';
import 'package:flutter_ui/core/models/userOperationClaim/userOperationClaimModel.dart';
import 'package:flutter_ui/core/services/service.dart';
import 'package:flutter_ui/core/services/serviceRepository.dart';
import 'package:flutter_ui/environments/api.dart';

class UserOperationClaimService extends Service
    with
        ServiceRepository<UserOperationClaimAddModel, UserOperationClaimModel> {
  Future<ResponseModel> add(
      UserOperationClaimAddModel userOperationClaimAddModel) async {
    var response = await httpClient.post(
      Uri.parse(API_URL + "useroperationclaims/add"),
      body: userOperationClaimAddModel.toJson(),
    );

    var jsonData = json.decode(response.body);
    return ResponseModel.fromJson(jsonData);
  }

  Future<ResponseModel> delete(DeleteModel deleteModel) async {
    var response = await httpClient.post(
      Uri.parse(API_URL + "useroperationclaims/delete"),
      body: deleteModel.toJson(),
    );

    var jsonData = json.decode(response.body);
    return ResponseModel.fromJson(jsonData);
  }

  Future<ResponseModel> update(
      UserOperationClaimModel userOperationClaimModel) async {
    var response = await httpClient.post(
      Uri.parse(API_URL + "useroperationclaims/update"),
      body: userOperationClaimModel.toJson(),
    );

    var jsonData = json.decode(response.body);
    return ResponseModel.fromJson(jsonData);
  }

  Future<ListResponseModel<UserOperationClaimModel>> getAll() async {
    var response = await httpClient.get(
      Uri.parse(API_URL + "useroperationclaims/getall"),
    );

    var jsonData = json.decode(response.body);
    return ListResponseModel<UserOperationClaimModel>.fromJson(jsonData);
  }

  Future<SingleResponseModel<UserOperationClaimModel>> getById(int id) async {
    var response = await httpClient.get(
      Uri.parse(API_URL + "useroperationclaims/getbyid?id=" + id.toString()),
    );

    var jsonData = json.decode(response.body);
    return SingleResponseModel<UserOperationClaimModel>.fromJson(jsonData);
  }

  Future<ListResponseModel<UserOperationClaimModel>> getByUserId(
      int userId) async {
    var response = await httpClient.get(
      Uri.parse(API_URL +
          "useroperationclaims/getbyuserid?userId=" +
          userId.toString()),
    );

    var jsonData = json.decode(response.body);
    return ListResponseModel<UserOperationClaimModel>.fromJson(jsonData);
  }

  Future<ListResponseModel<UserOperationClaimModel>> getByClaimId(
      int claimId) async {
    var response = await httpClient.get(
      Uri.parse(API_URL +
          "useroperationclaims/getbyclaimid?claimId=" +
          claimId.toString()),
    );

    var jsonData = json.decode(response.body);
    return ListResponseModel<UserOperationClaimModel>.fromJson(jsonData);
  }
}

File: AngularUI/angular-ui/src/app/core/services/user-operation-claim.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { apiUrl } from 'src/api';
import { DeleteModel } from '../models/deleteModel';
import { ListResponseModel } from '../models/response/listResponseModel';
import { ResponseModel } from '../models/response/responseModel';
import { SingleResponseModel } from '../models/response/singleResponseModel';
import { UserOperationClaimAddModel } from '../models/user-operation-claim/userOperationClaimAddModel';
import { UserOperationClaimModel } from '../models/user-operation-claim/userOperationClaimModel';
import { ServiceRepositoryWithAddModel } from './service-repository';

@Injectable({
  providedIn: 'root',
})
export class UserOperationClaimService
  implements
    ServiceRepositoryWithAddModel<
      UserOperationClaimAddModel,
      UserOperationClaimModel,
      number
    >
{
  constructor(private httpClient: HttpClient) {}

  add(
    userOperationClaimAddModel: UserOperationClaimAddModel
  ): Observable<ResponseModel> {
    let newPath = apiUrl + 'useroperationclaims/add';
    return this.httpClient.post<ResponseModel>(
      newPath,
      userOperationClaimAddModel
    );
  }

  delete(deleteModel: DeleteModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'useroperationclaims/delete';
    return this.httpClient.post<ResponseModel>(newPath, deleteModel);
  }

  update(
    userOperationClaimModel: UserOperationClaimModel
  ): Observable<ResponseModel> {
    let newPath = apiUrl + 'useroperationclaims/update';
    return this.httpClient.post<ResponseModel>(
      newPath,
      userOperationClaimModel
    );
  }

  getAll(): Observable<ListResponseModel<UserOperationClaimModel>> {
    let newPath = apiUrl + 'useroperationclaims/getall';
    return this.httpClient.get<ListResponseModel<UserOperationClaimModel>>(
      newPath
    );
  }

  getByUserId(
    userId: number
  ): Observable<ListResponseModel<UserOperationClaimModel>> {
    let newPath = apiUrl + 'useroperationclaims/getbyuserid?userId=' + userId;
    return this.httpClient.get<ListResponseModel<UserOperationClaimModel>>(
      newPath
    );
  }

  getByClaimId(
    claimId: number
  ): Observable<ListResponseModel<UserOperationClaimModel>> {
    let newPath =
      apiUrl + 'useroperationclaims/getbyclaimid?claimId=' + claimId;
    return this.httpClient.get<ListResponseModel<UserOperationClaimModel>>(
      newPath
    );
  }

  getById(
    id: number
  ): Observable<SingleResponseModel<UserOperationClaimModel>> {
    let newPath = apiUrl + 'useroperationclaims/getbyid?id=' + id;
    return this.httpClient.get<SingleResponseModel<UserOperationClaimModel>>(
      newPath
    );
  }
}
